### PR TITLE
Course

### PR DESCRIPTION
**Pull request by shashi-99**

### CODE REVIEW

File: src/service/assignmentService.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    public async deleteAssignment(id: number) {
+        try {
+            const affectedRows = await Assignment.destroy({
+                where: {
+                    assignment_id: id
+                }
+            });
+            if (affectedRows === 0) {
+                throw new Error();
```
Comment (saikat93) [inline src/service/assignmentService.ts]: You need to throw some error msg!

File: src/service/assignmentService.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            }
+            return successEnums.DELETE_SUCCESS;
+        } catch (error) {
+            logger.error(error);
+            logger.error(errorEnums.INT_SERVER_ERR);
```
Comment (saikat93) [inline src/service/assignmentService.ts:60]: Try to log one time.

File: src/service/assignmentService.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    public async updateAssignment(id: number, updates: Object) {
+        try {
+            const [affectedRows] = await Assignment.update(updates, {
+                where: {
+                    assignment_id: id
+                }
+            })
+            if (affectedRows === 0) {
+                logger.error(errorEnums.ERR_INVALID_INPUT);
```
Comment (saikat93) [inline src/service/assignmentService.ts:72]: Before you are throwing now you are logging. ? Follow one

**Review by saikat93 — APPROVED** [review]

File: src/service/courseService.ts [hunk 4]
```diff
@@ -44,11 +48,10 @@ class CourseService {
                 }
             });
             if (affectedRows === 0) {
-                throw new Error();
+                logger.error(errorEnums.ERR_INVALID_INPUT);
             }
             return successEnums.DELETE_SUCCESS;
```
Comment (saikat93) [inline src/service/courseService.ts]: In linne no 51 you are logging error and then line 53 you are returning success. Correct here and if you see in other place also.

File: src/service/courseService.ts [hunk 5]
```diff
@@ -11,16 +11,20 @@ class CourseService {
...
     }
 
     public async getCourseById(id: number) {
-        const course = await Course.findByPk(id);
-        if (!course) {
-            logger.error(errorEnums.ERR_INVALID_INPUT);
+        try {
+            const course = await Course.findByPk(id);
+            if (!course) {
+                logger.error(errorEnums.ERR_INVALID_INPUT);
```
Comment (saikat93) [inline src/service/courseService.ts:21]: same here

File: src/service/assignmentService.ts (hunk 2 above)
Comment (shashi-99) [reply src/service/assignmentService.ts:60]: Done!

**Comment by boring-anime** [comment]
The branch name is not revealing it's intention, This PR is doing multiple things which is breaking the single responsibility principle

**Pull request closed without merging**